app/safecast/tasks/device_story/update_metadata.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Tasks
+  module DeviceStory
+    class UpdateMetadata
+      class << self
+        def call
+          new.call
+        end
+      end
+
+      def call
+        each_metadata(parse(fetch(devices_uri))) do |device_urn, metadata|
+          begin
+            find_or_update_device(device_urn, metadata)
+          rescue ActiveRecord::RecordInvalid
+            warn "Could not update device metadata for #{device_urn}."
+          end
+        end
+      end
+
+      private
+
+      def find_or_update_device(device_urn, metadata)
+        device = ::DeviceStory::Device.find_by(device_urn: device_urn)
+        if device
+          device.update!(metadata: metadata)
+        else
+          ::DeviceStory::Device.create!(device_urn: device_urn, metadata: metadata)
```

Comment:
If we move to Rails 6 we can use `upsert` which is much more reliable in multi-threaded environments :)